Ariakit: Update to v0.4.10

tyxla opened the pull request:
## What?
Updates `@ariakit/react` to the latest `v0.4.10`.

I'm planning to use the new APIs in subsequent PRs.

## Why?
Mostly to consider a few fixes and the new `useStoreState` API that helps with React Compiler compatibility. Will be done in another PR. 

## How?
Just updating the dependency and resolving test failures where necessary.

## Testing Instructions
* Smoke all Ariakit components (including `Composite`, `CustomSelectControl`, `DropdownMenuV2`, `RadioGroup`, `TabPanel`, `Tabs`, `ToggleGroupControl`, `Tooltip`, `Toolbar`)
* Verify all checks are green.

### Testing Instructions for Keyboard
Verify all the above components are navigable with keyboard navigation like before.

## Screenshots or screencast <!-- if applicable -->

tyxla committed:
Ariakit: Update to v0.4.9

ciampo reviewed (commented):
Things are looking good after an initial smoke test.

Should we migrate `store.useState( stateProp )`  to `useStoreState(store, stateProp)` (new 0.4.9 feature), or is it unnecessary / premature?

tyxla commented:
> Should we migrate `store.useState( stateProp )` to `useStoreState(store, stateProp)` (new 0.4.9 feature), or is it unnecessary / premature?

I'm planning to do it in a follow-up PR.

tyxla committed:
CHANGELOG

ciampo commented on package-lock.json:
```diff
@@ -67558,8 +67558,7 @@
 				"@ariakit/react": {
-					"version": "0.4.7",
-					"resolved": "https://registry.npmjs.org/@ariakit/react/-/react-0.4.7.tgz",
+					"version": "https://registry.npmjs.org/@ariakit/react/-/react-0.4.7.tgz",
```
This new version of `package-lock.json` still points at a `0.4.7` file, is this expected?

tyxla replied:
Nope, this isn't right, I'll take a look

tyxla replied:
Should be resolved by b937a27cb580d5eb63b462a97adbe8fa6503ddf6. 

There was a dependency duplication going on, and I addressed it.

tyxla committed:
Deduplicate some ariakit dependencies

tyxla commented:
Now that the dependency was correctly updated, we'll see some failures related with the `v0.4.8` tab changes, I'll take a look at them.

ciampo commented:
I think that adding `accessibleWhenDisabled` to `Tabs.Tab` could solve the issue

tyxla commented:
> Adding `accessibleWhenDisabled` to `Tabs.Tab` seems to solve the issue on my local machine — although I thought that ariakit's `Tab` [already defaulted it to `true`](https://ariakit.org/reference/tab#accessiblewhendisabled) 🤔

Yup, seems like it might be coming from here: https://github.com/ariakit/ariakit/pull/4018

tyxla committed:
TabPanel: keep tabs accessible when disabled

ciampo commented:
> > Adding `accessibleWhenDisabled` to `Tabs.Tab` seems to solve the issue on my local machine — although I thought that ariakit's `Tab` [already defaulted it to `true`](https://ariakit.org/reference/tab#accessiblewhendisabled) 🤔
> 
> Yup, seems like it might be coming from here: [ariakit/ariakit#4018](https://github.com/ariakit/ariakit/pull/4018)

I'm surprised that the changed you linked affected us, since we never set `accessibleWhenDisabled` to `false` explicitly, and we instead relied on the fact that `accessibleWhenDisabled` has an implicit value of `true`.

Basically, it's unexpected to me that we have to set it explicitly, since I thought it already always defaulted to `true`.

ciampo commented on packages/components/src/tab-panel/index.tsx:
```diff
@@ -220,6 +220,7 @@ const UnforwardedTabPanel = (
 									showTooltip={ !! tab.icon }
 								/>
 							}
+							accessibleWhenDisabled
```
We may to have to do the same for the `Tab` component

```diff
diff --git a/packages/components/src/tabs/tab.tsx b/packages/components/src/tabs/tab.tsx
index e1aa85c636..8ab67e1ca9 100644
--- a/packages/components/src/tabs/tab.tsx
+++ b/packages/components/src/tabs/tab.tsx
@@ -26,6 +26,7 @@ export const Tab = forwardRef<
 	const instancedTabId = `${ instanceId }-${ tabId }`;
 	return (
 		<StyledTab
+			accessibleWhenDisabled
 			ref={ ref }
 			store={ store }
 			id={ instancedTabId }

```

tyxla replied:
Done in 312804105575044a83d6086268ec50cee4da26bb.

tyxla commented:
> > > Adding `accessibleWhenDisabled` to `Tabs.Tab` seems to solve the issue on my local machine — although I thought that ariakit's `Tab` [already defaulted it to `true`](https://ariakit.org/reference/tab#accessiblewhendisabled) 🤔
> > 
> > 
> > Yup, seems like it might be coming from here: [ariakit/ariakit#4018](https://github.com/ariakit/ariakit/pull/4018)
> 
> I'm surprised that the changed you linked affected us, since we never set `accessibleWhenDisabled` to `false` explicitly, and we instead relied on the fact that `accessibleWhenDisabled` has an implicit value of `true`.
> 
> Basically, it's unexpected to me that we have to set it explicitly, since I thought it already always defaulted to `true`.
> 
> (cc @diegohaz in case he can clarify why this is happening)

Agreed it's surprising - it is also documented to be `true` by default: https://ariakit.org/reference/tab#accessiblewhendisabled

tyxla committed:
Tab: enable accessibleWhenDisabled specifically

tyxla commented:
Note that ariakit/ariakit#4018 seems to be specifically removing the default `true` of `accessibleWhenDisabled`:

https://github.com/ariakit/ariakit/pull/4018/files#diff-1b1635ca34878fe6ca473aeecac2f52b9aaa31af0ed287d9c2b9ec14c5ebcf64L40

diegohaz commented:
> Note that [ariakit/ariakit#4018](https://github.com/ariakit/ariakit/pull/4018) seems to be specifically removing the default `true` of `accessibleWhenDisabled`:
> 
> https://github.com/ariakit/ariakit/pull/4018/files#diff-1b1635ca34878fe6ca473aeecac2f52b9aaa31af0ed287d9c2b9ec14c5ebcf64L40

Yes! 🤦🏽‍♂️

That was unintentional.

tyxla commented:
Here's a quick PR to reverse it with as little changes as needed: https://github.com/ariakit/ariakit/pull/4056 cc @diegohaz

tyxla commented:
Currently blocked by https://github.com/ariakit/ariakit/pull/4056

Will resume when there's a minor version that fixes the default of `accessibleWhenDisabled`.

tyxla committed:
Bump to 0.4.10

tyxla committed:
Revert "TabPanel: keep tabs accessible when disabled"

This reverts commit 44ede08ef4976d1f683690fd504fa78c833d0c30.

tyxla committed:
Revert "Tab: enable accessibleWhenDisabled specifically"

This reverts commit 312804105575044a83d6086268ec50cee4da26bb.

tyxla commented:
OK, I've updated to the latest version with the `accessibleWhenDisabled` fix and reverted the `accessibleWhenDisabled` patches. Should be ready for another look!

ciampo commented on packages/components/CHANGELOG.md:
```diff
@@ -68,10 +68,8 @@
-### Internal
-
 -   `CustomSelectControl`: Improve type inferring ([#64412](https://github.com/WordPress/gutenberg/pull/64412)).
+-   Update `ariakit` to version `0.4.9` ([#64637](https://github.com/WordPress/gutenberg/pull/64637)).
```
```suggestion
-   Update `ariakit` to version `0.4.10` ([#64637](https://github.com/WordPress/gutenberg/pull/64637)).
```

ciampo reviewed (approved):
🚀

tyxla committed:
Fix CHANGELOG

Co-authored-by: Marco Ciampini <marco.ciampo@gmail.com>
